templates/lib/prism/node.rb.erb

```diff
@@ -98,6 +98,9 @@ module Prism
       raise NoMethodError, "undefined method `type' for #{inspect}"
     end
   end
+  <%- if ENV["CHECK_FIELD_KIND"] -%>
+  CHECK_FIELD_KIND = ENV["CHECK_FIELD_KIND"]
```

Comment:
It's a build-time + run-time check to make it still possible to run e.g. `bin/parse -e 'code'` when investigating a type failure. If it was build-time only, that would require to `rm lib/prism/node.rb && bundle exec rake compile && bin/parse`.

Reply:
I think it could make it easier to test as well.

Reply:
I agree it's verbose in this template.
Generating another file seems quite heavy though, and we probably wouldn't want anything to use these checks besides that specific test added in CI (because it would be a lot slower).
I'm thinking to add some helper methods in template.rb to make this shorter and more readable, then it would only be a couple lines added in this file.

Reply:
Done, I simplified this a lot in the last commit.